### PR TITLE
Adds a golden allTypes for proto3

### PR DESCRIPTION
**Pull request by oldergod**
It'll be easier to iterate over fixing things.

Commented out things are broken now, will fix in follow up of course!

### CODE REVIEW

File: wire-library/wire-tests/src/commonTest/proto-kotlin/com/squareup/wire/protos3/kotlin/all_types/AllTypes.kt [hunk 1]
```diff
@@ -0,0 +1,1390 @@
+// Code generated by Wire protocol buffer compiler, do not edit.
```
Comment (swankjesse) [inline wire-library/wire-tests/src/commonTest/proto-kotlin/com/squareup/wire/protos3/kotlin/all_types/AllTypes.kt:1]: this package is protos3. Should it be protos3 or proto3? I think we use both and should not!

File: wire-library/wire-tests/src/commonTest/proto3/kotlin/all_types.proto [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+  repeated NestedEnum pack_nested_enum = 316 [packed = true];
+//  repeated google.protobuf.NullValue pack_null_value = 323 [packed = true];
+
+  map<int32, int32> map_int32_int32 = 501;
+  map<string, string> map_string_string = 502;
+  map<string, NestedMessage> map_string_message = 503;
+  map<string, NestedEnum> map_string_enum = 504;
+  map<int32, google.protobuf.Any> map_int32_any = 518;
+  map<int32, google.protobuf.Duration> map_int32_duration = 519;
+//  map<int32, google.protobuf.Struct> map_int32_struct = 520;
```
Comment (swankjesse) [inline wire-library/wire-tests/src/commonTest/proto3/kotlin/all_types.proto:106]: commented out why?

Comment (swankjesse) [reply wire-library/wire-tests/src/commonTest/proto3/kotlin/all_types.proto:106]: presumably it’s broken?!

Comment (swankjesse) [reply wire-library/wire-tests/src/commonTest/proto3/kotlin/all_types.proto:106]: ahhh I see it on your PR note! SGTM

File: wire-library/wire-tests/src/commonTest/proto-kotlin/com/squareup/wire/protos3/kotlin/all_types/AllTypes.kt (hunk 1 above)
Comment (oldergod) [reply wire-library/wire-tests/src/commonTest/proto-kotlin/com/squareup/wire/protos3/kotlin/all_types/AllTypes.kt]: fixed

File: wire-library/wire-tests/src/commonTest/proto3/kotlin/all_types.proto (hunk 2 above)
Comment (oldergod) [reply wire-library/wire-tests/src/commonTest/proto3/kotlin/all_types.proto:106]: The generated code doesn't compile yes

**Pull request merged**